extensions/amp-gist/0.1/amp-gist.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds a Github gist
+ *
+ * Example:
+ * <code>
+ * <amp-gist
+ *   layout="responsive"
+ *   data-gistid="a19e811dcd7df10c4da0931641538497"
+ *   width="100"
+ *   height="100">
+ * </amp-gist>
+ * </code>
+ */
+
+import {getIframe} from '../../../src/3p-frame';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+export class AmpGist extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+  }
+
+  /**
+  * @param {boolean=} opt_onLayout
+  * @override
+  */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://gist.github.com/', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**@override*/
+  layoutCallback() {
+    const gistid = user().assert(
+      this.element.getAttribute('data-gistid'),
+      'The data-gistid attribute is required for <amp-gist> %s',
+      this.element);
+    const width = this.element.getAttribute('width');
+    const height = this.element.getAttribute('height');
+    const url = 'https://gist.github.com/';
+
+    const iframe = getIframe(this.win, this.element, 'gist');
+
+    iframe.setAttribute('frameborder', 'no');
+    iframe.setAttribute('scrolling', 'no');
+
+    const src = url + encodeURIComponent(gistid) + '.pibb';
+
+    iframe.src = src;
+
+    this.applyFillContent(iframe);
+    iframe.width = width;
```

Comment:
@aghassemi I believe I was able to figure it out. I did create a new branch on my fork to do some trial/error stuff.

In my `3p/github.js` I got the size of the document by doing the following. I wasn't sure how else to do this. It appears `facebook.js` has access to the FB SDK, which reports the height of the document.

```js
const ctnr = document./*REVIEW*/querySelector('#c .gist');
```

I also used `requestResize` instead of `updateDimensions` because I couldn't get that to work.

```js
// . . .
window.context.requestResize(
  container./*REVIEW*/offsetWidth,
  ctnr./*REVIEW*/offsetHeight + /* margins */ 20
);
// . . .
```

I also removed one of my tests that was looking for the `.pibb` file from how I did it before. I think this may need a few more tests, but I wanted to go ahead and get you this, to see if I headed in the right direction. 

If I am way off base, please let me know, and I'll do it the simpler way for now. Thanks again for your help :)